### PR TITLE
KEB: Implement upgrade for CLS

### PR DESCRIPTION
**Pull request by suleymanakbas91**
**Description**

Changes proposed in this pull request:

- KEB: Implement upgrade for CLS

### CODE REVIEW

**Review by skhalash — APPROVED** [review]
LGTM!

File: components/kyma-environment-broker/internal/process/provisioning/cls_audit_log_step.go [hunk 1]
```diff
@@ -54,77 +49,35 @@ func (alo *CLSAuditLogOverrides) Run(operation internal.ProvisioningOperation, l
 	replaceSubAccountID := strings.Replace(string(luaScript), "sub_account_id", operation.ProvisioningParameters.ErsContext.SubAccountID, -1)
 	replaceTenantID := strings.Replace(replaceSubAccountID, "tenant_id", alo.auditLogConfig.Tenant, -1)
 
-	u, err := url.Parse(alo.auditLogConfig.URL)
+	auditlogOverrideParams, err := auditlog.PrepareOverrideParams(&alo.auditLogConfig, alo.secretKey, operation.Cls.Overrides)
 	if err != nil {
-		logger.Errorf("Unable to parse the URL: %v", err.Error())
+		logger.Errorf("Unable to prepare Audit Log override parameters: %v", err)
```
Comment (piotrmiskiewicz) [inline components/kyma-environment-broker/internal/process/provisioning/cls_audit_log_step.go:54]: if you know that there is no sense to retry operation, please make it failed:
```return s.operationManager.OperationFailed(operation, msg)```

File: components/kyma-environment-broker/internal/process/upgrade_kyma/cls_audit_log.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+		return operation, 0, err
+	}
+
+	replaceSubAccountID := strings.Replace(string(luaScript), "sub_account_id", operation.ProvisioningParameters.ErsContext.SubAccountID, -1)
+	replaceTenantID := strings.Replace(replaceSubAccountID, "tenant_id", alo.auditLogConfig.Tenant, -1)
+
+	auditlogOverrideParams, err := auditlog.PrepareOverrideParams(&alo.auditLogConfig, alo.secretKey, operation.Cls.Overrides)
+	if err != nil {
+		logger.Errorf("Unable to prepare Audit Log override parameters: %v", err)
+		return operation, 0, err
```
Comment (piotrmiskiewicz) [inline components/kyma-environment-broker/internal/process/upgrade_kyma/cls_audit_log.go:55]: please make the operation failed (and also in other cases): return s.operationManager.OperationFailed(operation, msg)

File: components/kyma-environment-broker/internal/process/upgrade_kyma/cls_offering.go [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+	}
+
+	smCredentials, err := cls.FindCredentials(s.config.ServiceManager, smRegion)
+	if err != nil {
+		return s.handleError(operation, err, err.Error(), log)
+	}
+
+	smClient := operation.SMClientFactory.ForCredentials(smCredentials)
+
+	meta, err := servicemanager.GenerateMetadata(smClient, provisioning.ClsOfferingName, provisioning.ClsPlanName)
```
Comment (piotrmiskiewicz) [inline components/kyma-environment-broker/internal/process/upgrade_kyma/cls_offering.go:53]: the code below also could be moved to the function.

File: components/kyma-environment-broker/internal/servicemanager/metadata.go [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+	kebError "github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/error"
+)
+
+type Metadata struct {
+	ServiceID string
+	BrokerID  string
+	PlanID    string
+}
+
+func GenerateMetadata(c Client, offeringName, planName string) (*Metadata, error) {
```
Comment (piotrmiskiewicz) [inline components/kyma-environment-broker/internal/servicemanager/metadata.go:15]: 'Metadata' does not say anything important, let's change the name and use "Offering" in that name

**Review by skhalash — APPROVED** [review]

**Pull request merged**